File: C/examples/test_HDC20X0.c
#include "rp-hdc20x0.h"
#include <stdio.h>
#include <unistd.h>

int main()
{
	float temperature, humidity; 

	if(setup_hdc20x0() != 0 ) return 0; 
    
	for (int i = 0; i < 5; i++) {
		if(read_from_hdc20x0(&temperature, &humidity)!=0) return 0;

		printf("{ \"temperature\" : %.1f, \"humidity\" : %.1f }\n", temperature, humidity);
		sleep(1);
	}
    
    if(hdc20x0_heater_on() != 0 ) {
        printf("Unable to turn heater on");
        return -3;
    }
    
    printf("Turned heater on. Temperature should climb up:\n");
    for (int i = 0; i < 5; i++) {
		if(read_from_hdc20x0(&temperature, &humidity)!=0) return 0;

		printf("{ \"temperature\" : %.1f, \"humidity\" : %.1f }\n", temperature, humidity);
		sleep(1);
	}
    
    if(hdc20x0_heater_off() != 0 ) {
        printf("Unable to turn heater on");
        return -3;
    }    
    
    printf("Turned heater off. Temperature should drop off:\n");
    for (int i = 0; i < 5; i++) {
		if(read_from_hdc20x0(&temperature, &humidity)!=0) return 0;

		printf("{ \"temperature\" : %.1f, \"humidity\" : %.1f }\n", temperature, humidity);
		sleep(1);
	}

	return 0;
}

File: Python/testHDC20X0.py
#!/usr/bin/python
from __future__ import print_function

import time
import hdc20x0


hdc20x0 = hdc20x0.Hdc20x0()
print("Sensor initialized")
print("{ temperature : %3.2f, humidity : %3.2f }" % (hdc20x0.readTemperature(), hdc20x0.readHumidity()))

print("Retrievieng a few values for control:")
for i in range(0, 5):
    print("{ temperature : %3.2f, humidity : %3.2f }" % (hdc20x0.readTemperature(), hdc20x0.readHumidity()))
    time.sleep(1)
    
print("Setting heater on")
hdc20x0.turnHeaterOn()

print("Temperature should climb up:")
for i in range(0, 5):
    print("{ temperature : %3.2f, humidity : %3.2f }" % (hdc20x0.readTemperature(), hdc20x0.readHumidity()))
    time.sleep(0.5)
    
print("Setting heater off")
hdc20x0.turnHeaterOff()

print("Temperature should go down:")
for i in range(0, 5):
    print("{ temperature : %3.2f, humidity : %3.2f }" % (hdc20x0.readTemperature(), hdc20x0.readHumidity()))
    time.sleep(0.5)
